### PR TITLE
RPC server/stdlib enhancements

### PR DESCRIPTION
**Pull request by mattaudesse**
Completes (with exception of docs):
 + Add `bigNumberToNumber` to real `stdlib`
 + Push `bigNumberify` into real `stdlib` to widen interface for
   `stdlib` functions
 + Ensure that all account and contract producing `stdlib` functions are
   properly wrapped in `makeACC` or `makeCTC` in server

Hopefully facilitates easier testing for:
 - `makeHandle` data structure replacement
 - Handle deletion functions
 - Error-handling

~To discuss/finish~:
 - ~Tighten up test spec descriptions(?)~
 - ~This probably won't work in CI as-is; use just ETH+Ganache? Extend CI script?~
 - ~`grep` source to ensure I've not orphaned anything~
 - ~Knock out test TODOs (I've made these very obvious + self-contained;
   can be done later if that makes more sense~ [Update: deferred to Trello \#1032])
 - ~Should we loosen typesigs (`BigNumber` -> `any`) on~:
   - ~`fundFromFaucet`?~ (Completed in 0494611ffbc294d0ef563e01b6ce7ae10e00c798)
   - ~`transfer`?~ (Completed in a4f98b281130f3081850180143a3e9464fc96541)
 - ~Document~:
   - ~`bigNumberToNumber`~
   - ~Any functions whose typesigs have been loosened~ (Completed in 76049dae57d21600bcf91d77c5056339153c57dc)

### CODE REVIEW

File: js/stdlib/Makefile [hunk 1]
```diff
@@ -31,7 +31,12 @@ build: package.json
 
 .PHONY: run
 run: build
-	docker-compose run stdlib
+	docker-compose run test-stdlib-ETH
+	docker-compose run test-stdlib-ALGO
+	docker-compose run test-stdlib-FAKE
+	docker-compose run test-rpc-server-ETH
+	docker-compose run test-rpc-server-ALGO
+	docker-compose run test-rpc-server-FAKE
```
Comment (jeapostrophe) [inline js/stdlib/Makefile:39]: I think my preference would be to make it so we use `./reach run` for testing and then we don't have to copy the Docker set up at all. (There would be a dummy `X.rsh` to get it to run the `X.mjs` test script)

File: js/stdlib/package.mo.json [hunk 2]
```diff
@@ -36,9 +36,13 @@
   "scripts": {
     "beautify": "js-beautify --indent-size=2 --end-with-newline --brace-style=collapse-preserve-inline --replace *.mjs",
     "format": "eslint --ignore-path .gitignore --ext .mjs --fix .",
-    "test": "node --experimental-modules ETH-test.mjs",
+    "test-stdlib-ETH": "node --experimental-modules ./test/stdlib-ETH.mjs",
+    "test-stdlib-ALGO": "node --experimental-modules ./test/stdlib-ALGO.mjs",
+    "test-stdlib-FAKE": "node --experimental-modules ./test/stdlib-FAKE.mjs",
+    "test-rpc-server-ETH": "node --experimental-modules ./test/rpc-server-ETH.mjs",
+    "test-rpc-server-ALGO": "node --experimental-modules ./test/rpc-server-ALGO.mjs",
+    "test-rpc-server-FAKE": "node --experimental-modules ./test/rpc-server-FAKE.mjs",
```
Comment (jeapostrophe) [inline js/stdlib/package.mo.json:44]: If doing the strategy I talked about above, a change would be that the Node package wouldn't really know about these tests

File: js/stdlib/Dockerfile [hunk 3]
```diff
@@ -19,6 +19,7 @@ COPY package.json package-lock.json /stdlib/
 COPY ts/* /stdlib/ts/
 COPY *.mjs /stdlib/
 COPY types/* /stdlib/types/
+COPY test/* /stdlib/test/
```
Comment (jeapostrophe) [inline js/stdlib/Dockerfile:22]: Similarly, these tests wouldn't be included in the actual package, but would just be our thing

**Comment by jeapostrophe** [comment]
> * Tighten up test spec descriptions(?)

Meh, no worries

> * This probably won't work in CI as-is; use just ETH+Ganache? Extend CI script?

Yes, although I believe that presently the CI does some sort of JS testing

> * `grep` source to ensure I've not orphaned anything

I didn't notice anything

> * Knock out test TODOs (I've made these very obvious + self-contained;
>   can be done later if that makes more sense)

I'd rather you make a card for this and put it in the INBOX

> * Should we loosen typesigs (`BigNumber` -> `any`) on:
>   
>   * `fundFromFaucet`?
>   * `transfer`?

Yes

> * Document:
>   
>   * `bigNumberToNumber`
>   * Any functions whose typesigs have been loosened

Yes, in this PR, please

File: js/stdlib/Makefile [hunk 4]
```diff
@@ -26,24 +30,24 @@ build: package.json
...
+#   to leverage the `reach` script and abstract over boilerplate such as
+#   `docker-compose.yml` config.
 .PHONY: test
-test: run
+test: build clean-test
+	mkdir test/stdlib-test
+	cp    test/index.* test/stdlib-test
+	cp -r test/lib     test/stdlib-test/lib
+	cd test/stdlib-test \
+	  && reach scaffold \
```
Comment (jeapostrophe) [inline js/stdlib/Makefile:45]: Since you are not patching it, this is not necessary

**Review by jeapostrophe — COMMENTED** [review]
Excellent!

File: js/stdlib/Makefile (hunk 4 above)
Comment (mattaudesse) [reply js/stdlib/Makefile]: Thanks - fixed in a4d0681f7c934591e14556e44afe4c0a9578ded9.

**Comment by mattaudesse** [comment]
> > * Tighten up test spec descriptions(?)
> 
> Meh, no worries
> 
> > * This probably won't work in CI as-is; use just ETH+Ganache? Extend CI script?
> 
> Yes, although I believe that presently the CI does some sort of JS testing
> 
> > * `grep` source to ensure I've not orphaned anything
> 
> I didn't notice anything
> 
> > * Knock out test TODOs (I've made these very obvious + self-contained;
> >   can be done later if that makes more sense)
> 
> I'd rather you make a card for this and put it in the INBOX
> 
> > * Should we loosen typesigs (`BigNumber` -> `any`) on:
> >   
> >   * `fundFromFaucet`?
> >   * `transfer`?
> 
> Yes
> 
> > * Document:
> >   
> >   * `bigNumberToNumber`
> >   * Any functions whose typesigs have been loosened
> 
> Yes, in this PR, please

I believe all the boxes have been ticked as of 76049dae57d21600bcf91d77c5056339153c57dc so I'll flip off the "WIP" now.

**Comment by jeapostrophe** [comment]
Great, this is good to rebase and merge

**Comment by mattaudesse** [comment]
Completed in 0fe67ea7bfcd260ce429cfd4ffac5c157a368093.

**Pull request closed without merging**